### PR TITLE
openai: allow sending user IDs

### LINKED ISSUE #612: Add Support for Sending End-user IDS to Openai. This will prevent Openai account suspention incase of misuse by end user.
Hi as referenced [here](https://platform.openai.com/docs/guides/safety-best-practices/end-user-ids) OpenAI accepts End user ids. It might be useful tool to help OpenAI monitor and detect abuse. This allows OpenAI to provide your team with more actionable feedback in the event that we detect any policy violations in your application.

### PR DESCRIPTION
**Pull request by nbsp**
Fixes #612

chose to add it to `chat()` but might make sense to add it to `LLM()`.

### CODE REVIEW

**Comment by davidzhao** [comment]
I think adding to LLM constructor would be better. VoiceAssistant will be issuing the `chat` calls and it wouldn't be passing in additional details such as this.

**Review by davidzhao — APPROVED** [review]

**Comment by davidzhao** [comment]
wanna fix the type check while you are there?

File: livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/llm.py [hunk 1]
```diff
@@ -212,13 +232,16 @@ def chat(
             if fnc_ctx and parallel_tool_calls is not None:
                 opts["parallel_tool_calls"] = parallel_tool_calls
 
+        user = self._opts.user or openai.NOT_GIVEN
```
Comment (theomonnom) [inline livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/llm.py:235]: is openai.NOT_GIVEN exported? last time I checked it wasn't

Comment (nbsp) [reply livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/llm.py:235]: https://github.com/openai/openai-python/blob/3bf03f2f7ee1a9e214484524000774532d02cba7/src/openai/__init__.py#L45

**Pull request merged**